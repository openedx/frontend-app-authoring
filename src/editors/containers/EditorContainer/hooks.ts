import { useState } from 'react';
import { useSelector } from 'react-redux';

import analyticsEvt from '../../data/constants/analyticsEvt';
import { RequestKeys } from '../../data/constants/requests';
import { selectors } from '../../data/redux';
import { StrictDict } from '../../utils';
import * as appHooks from '../../hooks';

export const {
  clearSaveError,
  navigateCallback,
  nullMethod,
  saveBlock,
} = appHooks;

export const state = StrictDict({
  // eslint-disable-next-line react-hooks/rules-of-hooks
  isCancelConfirmModalOpen: (val) => useState(val),
});

export const handleSaveClicked = ({
  dispatch,
  getContent,
  validateEntry,
  returnFunction,
}) => {
  // eslint-disable-next-line react-hooks/rules-of-hooks
  const returnUrl = useSelector(selectors.app.returnUrl);
  const destination = returnFunction ? '' : returnUrl;
  // eslint-disable-next-line react-hooks/rules-of-hooks
  const analytics = useSelector(selectors.app.analytics);

  return () => saveBlock({
    analytics,
    content: getContent({ dispatch }),
    destination,
    dispatch,
    returnFunction,
    validateEntry,
  });
};

export const cancelConfirmModalToggle = () => {
  const [isCancelConfirmOpen, setIsOpen] = state.isCancelConfirmModalOpen(false);
  return {
    isCancelConfirmOpen,
    openCancelConfirmModal: () => setIsOpen(true),
    closeCancelConfirmModal: () => setIsOpen(false),
  };
};

export const handleCancel = ({
  onClose = null,
  returnFunction = null,
}: {
  onClose?: (() => void) | null;
  returnFunction?: (() => (result: any) => void) | null;
}): ((result?: any) => void) => {
  if (onClose) {
    return onClose;
  }
  // eslint-disable-next-line react-hooks/rules-of-hooks
  const returnUrl = useSelector(selectors.app.returnUrl);
  return navigateCallback({
    returnFunction,
    // eslint-disable-next-line react-hooks/rules-of-hooks
    destination: returnFunction ? '' : returnUrl,
    analyticsEvent: analyticsEvt.editorCancelClick,
    // eslint-disable-next-line react-hooks/rules-of-hooks
    analytics: useSelector(selectors.app.analytics),
  });
};

// eslint-disable-next-line react-hooks/rules-of-hooks
export const isInitialized = () => useSelector(selectors.app.isInitialized);

// eslint-disable-next-line react-hooks/rules-of-hooks
export const saveFailed = () => useSelector((rootState) => (
  selectors.requests.isFailed(rootState, { requestKey: RequestKeys.saveBlock })
));
