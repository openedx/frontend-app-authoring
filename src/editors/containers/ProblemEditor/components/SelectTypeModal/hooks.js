import { useEffect } from 'react';
import {
  AdvanceProblemKeys, AdvanceProblems, ProblemTypeKeys, ProblemTypes,
} from '../../../../data/constants/problem';
import { snakeCaseKeys } from '../../../../utils';
import { getDataFromOlx } from '../../../../data/redux/thunkActions/problem';

export const onSelect = ({
  selected,
  updateField,
  setBlockTitle,
  defaultSettings,
}) => () => {
  if (Object.values(AdvanceProblemKeys).includes(selected)) {
    updateField({ problemType: ProblemTypeKeys.ADVANCED, rawOLX: AdvanceProblems[selected].template });
    setBlockTitle(AdvanceProblems[selected].title);
  } else {
    const newOLX = ProblemTypes[selected].template;
    const newState = getDataFromOlx({
      rawOLX: newOLX,
      rawSettings: {
        weight: 1,
        attempts_before_showanswer_button: 0,
        show_reset_button: null,
        showanswer: null,
        defaultToAdvanced: false,
      },
      defaultSettings: snakeCaseKeys(defaultSettings),
    });
    updateField(newState);
    setBlockTitle(ProblemTypes[selected].title);
  }
};

export const useArrowNav = (selected, setSelected) => {
  const detectKeyDown = (e) => {
    const problemTypeValues = Object.values(ProblemTypeKeys);
    switch (e.key) {
      case 'ArrowUp':
        if (problemTypeValues.includes(selected) && ProblemTypes[selected].prev) {
          setSelected(ProblemTypes[selected].prev);
          document.getElementById(ProblemTypes[selected].prev).focus();
        }
        break;
      case 'ArrowDown':
        if (problemTypeValues.includes(selected) && ProblemTypes[selected].next) {
          setSelected(ProblemTypes[selected].next);
          document.getElementById(ProblemTypes[selected].next).focus();
        }
        break;
      default:
    }
  };
  useEffect(() => {
    document.addEventListener('keydown', detectKeyDown, true);
    return () => {
      document.removeEventListener('keydown', detectKeyDown, true);
    };
  }, [selected, setSelected]);
};
