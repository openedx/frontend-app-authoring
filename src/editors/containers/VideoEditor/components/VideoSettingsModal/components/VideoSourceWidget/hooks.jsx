import React from 'react';
import { actions } from '../../../../../../data/redux';
import { parseYoutubeId } from '../../../../../../data/services/cms/api';
import * as requests from '../../../../../../data/redux/thunkActions/requests';

export const state = {
  // eslint-disable-next-line react-hooks/rules-of-hooks
  showVideoIdChangeAlert: (args) => React.useState(args),
};

export const sourceHooks = ({ dispatch, previousVideoId, setAlert }) => ({
  updateVideoURL: (e, videoId) => {
    const videoUrl = e.target.value;
    dispatch(actions.video.updateField({ videoSource: videoUrl }));

    const youTubeId = parseYoutubeId(videoUrl);
    if (youTubeId) {
      dispatch(requests.checkTranscriptsForImport({
        videoId,
        youTubeId,
        onSuccess: (response) => {
          if (response.data.command === 'import') {
            dispatch(actions.video.updateField({
              allowTranscriptImport: true,
            }));
          }
        },
      }));
    }
  },
  updateVideoId: (e) => {
    const updatedVideoId = e.target.value;
    if (previousVideoId !== updatedVideoId && updatedVideoId) {
      setAlert();
    }
    dispatch(actions.video.updateField({ videoId: updatedVideoId }));
  },
});

export const fallbackHooks = ({ fallbackVideos, dispatch }) => ({
  addFallbackVideo: () => dispatch(actions.video.updateField({ fallbackVideos: [...fallbackVideos, ''] })),

  deleteFallbackVideo: (videoIndex) => {
    const updatedFallbackVideos = fallbackVideos.reduce((result, currentItem, index) => {
      if (index === videoIndex) { return result; }
      return [...result, currentItem];
    }, []);

    dispatch(actions.video.updateField({ fallbackVideos: updatedFallbackVideos }));
  },
});

export const videoIdChangeAlert = () => {
  const [showVideoIdChangeAlert, setShowVideoIdChangeAlert] = state.showVideoIdChangeAlert(false);
  return {
    videoIdChangeAlert: {
      show: showVideoIdChangeAlert,
      set: () => setShowVideoIdChangeAlert(true),
      dismiss: () => setShowVideoIdChangeAlert(false),
    },
  };
};

export default {
  videoIdChangeAlert,
  sourceHooks,
  fallbackHooks,
};
