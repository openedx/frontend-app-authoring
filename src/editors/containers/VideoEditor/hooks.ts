import { useState, createContext } from 'react';
import { thunkActions } from '../../data/redux';

import { StrictDict } from '../../utils';

type ErrRecord = Record<string, string>;
type ErrCategory = [errors: ErrRecord, setter: (newErrors: ErrRecord) => void];
interface ErrorContextData {
  duration: ErrCategory;
  handout: ErrCategory;
  license: ErrCategory;
  thumbnail: ErrCategory;
  transcripts: ErrCategory;
  videoSource: ErrCategory;
}

/* istanbul ignore next */
export const ErrorContext = createContext<ErrorContextData>({
  duration: [{}, () => {}],
  handout: [{}, () => {}],
  license: [{}, () => {}],
  thumbnail: [{}, () => {}],
  transcripts: [{}, () => {}],
  videoSource: [{}, () => {}],
});

export const state = StrictDict({
  /* eslint-disable react-hooks/rules-of-hooks */
  durationErrors: (val) => useState(val),
  handoutErrors: (val) => useState(val),
  licenseErrors: (val) => useState(val),
  thumbnailErrors: (val) => useState(val),
  transcriptsErrors: (val) => useState(val),
  videoSourceErrors: (val) => useState(val),
  /* eslint-enable react-hooks/rules-of-hooks */
});

export const errorsHook = (): { error: ErrorContextData, validateEntry: () => boolean } => {
  const [durationErrors, setDurationErrors] = state.durationErrors({});
  const [handoutErrors, setHandoutErrors] = state.handoutErrors({});
  const [licenseErrors, setLicenseErrors] = state.licenseErrors({});
  const [thumbnailErrors, setThumbnailErrors] = state.thumbnailErrors({});
  const [transcriptsErrors, setTranscriptsErrors] = state.transcriptsErrors({});
  const [videoSourceErrors, setVideoSourceErrors] = state.videoSourceErrors({});

  return {
    error: {
      duration: [durationErrors, setDurationErrors],
      handout: [handoutErrors, setHandoutErrors],
      license: [licenseErrors, setLicenseErrors],
      thumbnail: [thumbnailErrors, setThumbnailErrors],
      transcripts: [transcriptsErrors, setTranscriptsErrors],
      videoSource: [videoSourceErrors, setVideoSourceErrors],
    },
    validateEntry: () => {
      if (Object.keys(durationErrors).length > 0) { return false; }
      if (Object.keys(handoutErrors).length > 0) { return false; }
      if (Object.keys(licenseErrors).length > 0) { return false; }
      if (Object.keys(thumbnailErrors).length > 0) { return false; }
      if (Object.keys(transcriptsErrors).length > 0) { return false; }
      if (Object.keys(videoSourceErrors).length > 0) { return false; }
      return true;
    },
  };
};

export const fetchVideoContent = () => ({ dispatch }) => (
  dispatch(thunkActions.video.saveVideoData())
);
