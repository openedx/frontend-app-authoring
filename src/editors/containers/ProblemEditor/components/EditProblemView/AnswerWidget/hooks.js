import { useState, useEffect } from 'react';
import { StrictDict } from '../../../../../utils';
// This 'module' self-import hack enables mocking during tests.
// See src/editors/decisions/0005-internal-editor-testability-decisions.md. The whole approach to how hooks are tested
// should be re-thought and cleaned up to avoid this pattern.
// eslint-disable-next-line import/no-self-import
import * as module from './hooks';
import { actions } from '../../../../../data/redux';
import { ProblemTypeKeys } from '../../../../../data/constants/problem';
import { fetchEditorContent } from '../hooks';

export const state = StrictDict({
  // eslint-disable-next-line react-hooks/rules-of-hooks
  isFeedbackVisible: (val) => useState(val),
});

export const removeAnswer = ({
  answer,
  dispatch,
}) => () => {
  dispatch(actions.problem.deleteAnswer({
    id: answer.id,
    correct: answer.correct,
    editorState: fetchEditorContent({ format: '' }),
  }));
};

export const setAnswer = ({ answer, hasSingleAnswer, dispatch }) => (payload) => {
  dispatch(actions.problem.updateAnswer({ id: answer.id, hasSingleAnswer, ...payload }));
};

export const setAnswerTitle = ({
  answer,
  hasSingleAnswer,
  dispatch,
  problemType,
}) => (updatedTitle) => {
  let title = updatedTitle;
  if ([ProblemTypeKeys.TEXTINPUT, ProblemTypeKeys.NUMERIC, ProblemTypeKeys.DROPDOWN].includes(problemType)) {
    title = updatedTitle.target.value;
  }
  dispatch(actions.problem.updateAnswer({ id: answer.id, hasSingleAnswer, title }));
};

export const setSelectedFeedback = ({ answer, hasSingleAnswer, dispatch }) => (value) => {
  if (value) {
    dispatch(actions.problem.updateAnswer({
      id: answer.id,
      hasSingleAnswer,
      selectedFeedback: value,
    }));
  }
};

export const setUnselectedFeedback = ({ answer, hasSingleAnswer, dispatch }) => (value) => {
  if (value) {
    dispatch(actions.problem.updateAnswer({
      id: answer.id,
      hasSingleAnswer,
      unselectedFeedback: value,
    }));
  }
};

export const useFeedback = (answer) => {
  const [isFeedbackVisible, setIsFeedbackVisible] = module.state.isFeedbackVisible(false);
  useEffect(() => {
    // Show feedback fields if feedback is present
    const isVisible = !!answer.selectedFeedback || !!answer.unselectedFeedback;
    setIsFeedbackVisible(isVisible);
  }, [answer]);

  const toggleFeedback = (open) => {
    // Do not allow to hide if feedback is added
    const { selectedFeedback, unselectedFeedback } = fetchEditorContent({ format: '' });

    if (!!selectedFeedback?.[answer.id] || !!unselectedFeedback?.[answer.id]) {
      setIsFeedbackVisible(true);
      return;
    }
    setIsFeedbackVisible(open);
  };
  return {
    isFeedbackVisible,
    toggleFeedback,
  };
};

export const isSingleAnswerProblem = (problemType) => (
  problemType === ProblemTypeKeys.DROPDOWN
);

export const useAnswerContainer = ({ answers, updateField }) => {
  useEffect(() => {
    let answerCount = 0;
    answers.forEach(answer => {
      if (answer.correct) {
        answerCount += 1;
      }
    });
    updateField({ correctAnswerCount: answerCount });
  }, []);
};

export default {
  state, removeAnswer, setAnswer, setAnswerTitle, useFeedback, isSingleAnswerProblem, useAnswerContainer,
};
