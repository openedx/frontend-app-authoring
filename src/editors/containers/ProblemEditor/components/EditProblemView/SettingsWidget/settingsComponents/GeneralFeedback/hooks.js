import { useState, useEffect } from 'react';
import isEmpty from 'lodash/isEmpty';
import messages from './messages';
// This 'module' self-import hack enables mocking during tests.
// See src/editors/decisions/0005-internal-editor-testability-decisions.md. The whole approach to how hooks are tested
// should be re-thought and cleaned up to avoid this pattern.
// eslint-disable-next-line import/no-self-import
import * as module from './hooks';

export const state = {
  // eslint-disable-next-line react-hooks/rules-of-hooks
  summary: (val) => useState(val),
};

export const generalFeedbackHooks = (generalFeedback, updateSettings) => {
  const [summary, setSummary] = module.state.summary({
    message: messages.noGeneralFeedbackSummary, values: {}, intl: true,
  });

  // eslint-disable-next-line react-hooks/rules-of-hooks
  useEffect(() => {
    if (isEmpty(generalFeedback)) {
      setSummary({ message: messages.noGeneralFeedbackSummary, values: {}, intl: true });
    } else {
      setSummary({
        message: generalFeedback,
        values: {},
        intl: false,
      });
    }
  }, [generalFeedback]);

  const handleChange = (event) => {
    updateSettings({ generalFeedback: event.target.value });
  };

  return {
    summary,
    handleChange,
  };
};
