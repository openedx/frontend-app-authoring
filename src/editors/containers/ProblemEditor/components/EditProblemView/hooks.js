import { useState } from 'react';
import 'tinymce';
import { StrictDict } from '../../../../utils';
import ReactStateSettingsParser from '../../data/ReactStateSettingsParser';
import ReactStateOLXParser from '../../data/ReactStateOLXParser';
import { setAssetToStaticUrl } from '../../../../sharedComponents/TinyMceWidget/hooks';
import { ProblemTypeKeys } from '../../../../data/constants/problem';

export const state = StrictDict({
  // eslint-disable-next-line react-hooks/rules-of-hooks
  isSaveWarningModalOpen: (val) => useState(val),
});

export const saveWarningModalToggle = () => {
  const [isSaveWarningModalOpen, setIsOpen] = state.isSaveWarningModalOpen(false);
  return {
    isSaveWarningModalOpen,
    openSaveWarningModal: () => setIsOpen(true),
    closeSaveWarningModal: () => setIsOpen(false),
  };
};

/** Checks if any tinymce editor in window is dirty */
export const checkIfEditorsDirty = () => {
  const EditorsArray = window.tinymce.editors;
  return Object.entries(EditorsArray).some(([id, editor]) => {
    if (Number.isNaN(parseInt(id, 10))) {
      if (!editor.isNotDirty) {
        return true;
      }
    }
    return false;
  });
};

export const fetchEditorContent = ({ format }) => {
  const editorObject = { hints: [] };
  const EditorsArray = window.tinymce.editors;
  Object.entries(EditorsArray).forEach(([id, editor]) => {
    if (Number.isNaN(parseInt(id, 10))) {
      if (id.startsWith('answer')) {
        const { answers } = editorObject;
        const answerId = id.substring(id.indexOf('-') + 1);
        editorObject.answers = { ...answers, [answerId]: editor.getContent({ format }) };
      } else if (id.includes('Feedback')) {
        const { selectedFeedback, unselectedFeedback, groupFeedback } = editorObject;
        const feedbackId = id.substring(id.indexOf('-') + 1);
        if (id.startsWith('selected')) {
          editorObject.selectedFeedback = { ...selectedFeedback, [feedbackId]: editor.getContent() };
        }
        if (id.startsWith('unselected')) {
          editorObject.unselectedFeedback = { ...unselectedFeedback, [feedbackId]: editor.getContent() };
        }
        if (id.startsWith('group')) {
          editorObject.groupFeedback = { ...groupFeedback, [feedbackId]: editor.getContent() };
        }
      } else if (id.startsWith('hint')) {
        const { hints } = editorObject;
        hints.push(editor.getContent());
      } else {
        editorObject[id] = editor.getContent();
      }
    }
  });
  return editorObject;
};

export const parseState = ({
  problem,
  isAdvanced,
  ref,
  lmsEndpointUrl,
}) => () => {
  const rawOLX = ref?.current?.state.doc.toString();
  const editorObject = fetchEditorContent({ format: '' });
  const reactOLXParser = new ReactStateOLXParser({ problem, editorObject });
  const reactSettingsParser = new ReactStateSettingsParser({ problem, rawOLX });
  const reactBuiltOlx = setAssetToStaticUrl({ editorValue: reactOLXParser.buildOLX(), lmsEndpointUrl });
  return {
    settings: isAdvanced ? reactSettingsParser.parseRawOlxSettings() : reactSettingsParser.getSettings(),
    olx: isAdvanced ? rawOLX : reactBuiltOlx,
  };
};

export const checkForNoAnswers = ({ openSaveWarningModal, problem }) => {
  const simpleTextAreaProblems = [ProblemTypeKeys.DROPDOWN, ProblemTypeKeys.NUMERIC, ProblemTypeKeys.TEXTINPUT];
  const editorObject = fetchEditorContent({ format: '' });
  const { problemType } = problem;
  const { answers } = problem;
  const answerTitles = simpleTextAreaProblems.includes(problemType) ? {} : editorObject.answers;

  const hasTitle = () => {
    const titles = [];
    answers.forEach(answer => {
      const title = simpleTextAreaProblems.includes(problemType) ? answer.title : answerTitles[answer.id];
      if (title.length > 0) {
        titles.push(title);
      }
    });
    if (titles.length > 0) {
      return true;
    }
    return false;
  };

  const hasCorrectAnswer = () => {
    let correctAnswer;
    answers.forEach(answer => {
      if (answer.correct) {
        const title = simpleTextAreaProblems.includes(problemType) ? answer.title : answerTitles[answer.id];
        if (title.length > 0) {
          correctAnswer = true;
        }
      }
    });
    if (correctAnswer) {
      return true;
    }
    return false;
  };

  if (problemType === ProblemTypeKeys.NUMERIC && !hasTitle()) {
    openSaveWarningModal();
    return true;
  }
  if (!hasCorrectAnswer()) {
    openSaveWarningModal();
    return true;
  }
  return false;
};

export const checkForSettingDiscrepancy = ({ problem, ref, openSaveWarningModal }) => {
  const rawOLX = ref?.current?.state.doc.toString();
  const reactSettingsParser = new ReactStateSettingsParser({ problem, rawOLX });
  const problemSettings = reactSettingsParser.getSettings();
  const rawOlxSettings = reactSettingsParser.parseRawOlxSettings();
  let isMismatched = false;

  Object.entries(rawOlxSettings).forEach(([key, value]) => {
    if (value !== problemSettings[key]) {
      isMismatched = true;
    }
  });

  if (isMismatched) {
    openSaveWarningModal();
    return true;
  }
  return false;
};

export const getContent = ({
  problemState,
  openSaveWarningModal,
  isAdvancedProblemType,
  editorRef,
  lmsEndpointUrl,
}) => {
  const problem = problemState;
  const hasNoAnswers = isAdvancedProblemType ? false : checkForNoAnswers({
    problem,
    openSaveWarningModal,
  });
  const hasMismatchedSettings = isAdvancedProblemType ? checkForSettingDiscrepancy({
    ref: editorRef,
    problem,
    openSaveWarningModal,
  }) : false;
  if (!hasNoAnswers && !hasMismatchedSettings) {
    const data = parseState({
      isAdvanced: isAdvancedProblemType,
      ref: editorRef,
      problem,
      lmsEndpointUrl,
    })();
    return data;
  }
  return null;
};
