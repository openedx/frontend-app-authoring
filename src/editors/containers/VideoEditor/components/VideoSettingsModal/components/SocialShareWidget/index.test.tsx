import React from 'react';
import {
  render, screen, initializeMocks,
  fireEvent,
} from '@src/testUtils';

import { formatMessage } from '../../../../../../testUtils';
import { SocialShareWidgetInternal as SocialShareWidget } from '.';
import * as hooks from './hooks';

jest.mock('../../../../../../data/redux', () => ({
  actions: {
    video: {
      updateField: jest.fn().mockName('actions.video.updateField'),
    },
  },
  selectors: {
    app: {
      isLibrary: jest.fn(state => ({ isLibrary: state })),
    },
    video: {
      allowVideoSharing: jest.fn(state => ({ allowVideoSharing: state })),
      videoSharingEnabledForAll: jest.fn(state => ({ videoSharingEnabledForAll: state })),
      videoSharingEnabledForCourse: jest.fn(state => ({ videoSharingEnabledForCourse: state })),
      videoSharingLearnMoreLink: jest.fn(state => ({ videoSharingLearnMoreLink: state })),
    },
  },
}));

describe('SocialShareWidget', () => {
  const props = {
    intl: { formatMessage },
    videoSharingEnabledForCourse: false,
    videoSharingEnabledForAll: false,
    isLibrary: false,
    allowVideoSharing: {
      level: 'block',
      value: false,
    },
    videoSharingLearnMoreLink: 'sOMeURl.cOM',
    updateField: jest.fn().mockName('args.updateField'),
  };

  beforeEach(() => {
    initializeMocks();
    jest.spyOn(hooks, 'useTrackSocialSharingChange').mockReturnValue(jest.fn());
  });

  describe('rendered with videoSharingEnabled false', () => {
    describe('with default props', () => {
      it('should return null', () => {
        const { container } = render(<SocialShareWidget {...props} />);
        const reduxWrapper = container.firstChild;
        expect(reduxWrapper?.textContent).toBe('');
      });
    });

    describe('with videoSharingEnabledForAll false and isLibrary true', () => {
      it('should return null', () => {
        const { container } = render(<SocialShareWidget {...props} isLibrary />);
        const reduxWrapper = container.firstChild;
        expect(reduxWrapper?.textContent).toBe('');
      });
    });

    describe('with videoSharingEnabledForCourse and isLibrary false and videoSharingEnabledForAll true', () => {
      it('should return null', () => {
        const { container } = render(<SocialShareWidget {...props} videoSharingEnabledForAll />);
        const reduxWrapper = container.firstChild;
        expect(reduxWrapper?.textContent).toBe('');
      });
    });
  });

  describe('rendered with videoSharingEnabled true', () => {
    describe('and allowVideoSharing value equals true', () => {
      describe(' with level equal to course', () => {
        it('should have setting location message', () => {
          render(<SocialShareWidget
            {...props}
            videoSharingEnabledForCourse
            allowVideoSharing={{
              level: 'course',
              value: true,
            }}
          />);
          expect(screen.getByText('Change this setting on the course outline page.')).toBeInTheDocument();
        });
        it('should have checkbox disabled prop equal true', () => {
          render(<SocialShareWidget
            {...props}
            videoSharingEnabledForCourse
            allowVideoSharing={{
              level: 'course',
              value: true,
            }}
          />);
          const checkbox = screen.getByRole('checkbox', { name: 'This video is shareable to social media' });
          expect(checkbox).toBeInTheDocument();
          expect(checkbox).toBeDisabled();
        });
      });
      describe(' with level equal to block', () => {
        it('should not have setting location message', () => {
          render(<SocialShareWidget
            {...props}
            videoSharingEnabledForCourse
            allowVideoSharing={{
              level: 'block',
              value: true,
            }}
          />);
          expect(screen.queryByText('Change this setting on the course outline page.')).not.toBeInTheDocument();
        });
        it('should not have override note', () => {
          render(<SocialShareWidget
            {...props}
            videoSharingEnabledForCourse
            allowVideoSharing={{
              level: 'block',
              value: true,
            }}
          />);
          expect(screen.queryByText('Note: This setting is overridden by the course outline page.')).not.toBeInTheDocument();
        });
        it('should have checkbox disabled prop equal false', () => {
          render(<SocialShareWidget
            {...props}
            videoSharingEnabledForCourse
            allowVideoSharing={{
              level: 'block',
              value: true,
            }}
          />);
          const checkbox = screen.getByRole('checkbox', { name: 'This video is shareable to social media' });
          expect(checkbox).toBeInTheDocument();
          expect(checkbox).not.toBeDisabled();
        });
      });
      describe('isLibrary equals true', () => {
        it('should not have setting location message', () => {
          render(<SocialShareWidget
            {...props}
            videoSharingEnabledForAll
            isLibrary
            allowVideoSharing={{
              level: 'block',
              value: true,
            }}
          />);
          expect(screen.queryByText('Change this setting on the course outline page.')).not.toBeInTheDocument();
        });
        it('should not have override note', () => {
          render(<SocialShareWidget
            {...props}
            videoSharingEnabledForAll
            isLibrary
            allowVideoSharing={{
              level: 'block',
              value: true,
            }}
          />);
          expect(screen.queryByText('Note: This setting is overridden by the course outline page.')).not.toBeInTheDocument();
        });
        it('should have checkbox disabled prop equal false', () => {
          render(<SocialShareWidget
            {...props}
            videoSharingEnabledForAll
            isLibrary
            allowVideoSharing={{
              level: 'block',
              value: true,
            }}
          />);
          const checkbox = screen.getByRole('checkbox', { name: 'This video is shareable to social media' });
          expect(checkbox).toBeInTheDocument();
          expect(checkbox).not.toBeDisabled();
        });
      });
      it('should have subtitle with text that reads Enabled', () => {
        render(<SocialShareWidget
          {...props}
          videoSharingEnabledForCourse
          allowVideoSharing={{
            level: 'block',
            value: true,
          }}
        />);
        expect(screen.getByText('Social Sharing')).toBeInTheDocument();
        fireEvent.click(screen.getByText('Social Sharing'));
        const subtitle = screen.getByText('Enabled');
        expect(subtitle).toBeInTheDocument();
      });
    });

    describe('and allowVideoSharing value equals false', () => {
      describe(' with level equal to course', () => {
        it('should have setting location message', () => {
          render(<SocialShareWidget
            {...props}
            videoSharingEnabledForCourse
            allowVideoSharing={{
              level: 'course',
              value: false,
            }}
          />);
          expect(screen.getByText('Change this setting on the course outline page.')).toBeInTheDocument();
        });
        it('should have checkbox disabled prop equal true', () => {
          render(<SocialShareWidget
            {...props}
            videoSharingEnabledForCourse
            allowVideoSharing={{
              level: 'course',
              value: false,
            }}
          />);
          const checkbox = screen.getByRole('checkbox', { name: 'This video is shareable to social media' });
          expect(checkbox).toBeInTheDocument();
          expect(checkbox).toBeDisabled();
        });
      });

      describe(' with level equal to block', () => {
        it('should not have setting location message', () => {
          render(<SocialShareWidget
            {...props}
            videoSharingEnabledForCourse
            allowVideoSharing={{
              level: 'block',
              value: false,
            }}
          />);
          expect(screen.queryByText('Change this setting on the course outline page.')).not.toBeInTheDocument();
        });
        it('should not have override note', () => {
          render(<SocialShareWidget
            {...props}
            videoSharingEnabledForCourse
            allowVideoSharing={{
              level: 'block',
              value: false,
            }}
          />);
          expect(screen.queryByText('Note: This setting is overridden by the course outline page.')).not.toBeInTheDocument();
        });
        it('should have checkbox disabled prop equal false', () => {
          render(<SocialShareWidget
            {...props}
            videoSharingEnabledForCourse
            allowVideoSharing={{
              level: 'block',
              value: false,
            }}
          />);
          const checkbox = screen.getByRole('checkbox', { name: 'This video is shareable to social media' });
          expect(checkbox).toBeInTheDocument();
          expect(checkbox).toBeEnabled();
        });
      });

      describe('isLibrary equals true', () => {
        it('should not have setting location message', () => {
          render(<SocialShareWidget
            {...props}
            videoSharingEnabledForAll
            isLibrary
            allowVideoSharing={{
              level: 'block',
              value: false,
            }}
          />);
          expect(screen.queryByText('Change this setting on the course outline page.')).not.toBeInTheDocument();
        });
        it('should not have override note', () => {
          render(<SocialShareWidget
            {...props}
            videoSharingEnabledForAll
            isLibrary
            allowVideoSharing={{
              level: 'block',
              value: false,
            }}
          />);
          expect(screen.queryByText('Note: This setting is overridden by the course outline page.')).not.toBeInTheDocument();
        });
        it('should have checkbox disabled prop equal false', () => {
          render(<SocialShareWidget
            {...props}
            videoSharingEnabledForAll
            isLibrary
            allowVideoSharing={{
              level: 'block',
              value: false,
            }}
          />);
          const checkbox = screen.getByRole('checkbox', { name: 'This video is shareable to social media' });
          expect(checkbox).toBeInTheDocument();
          expect(checkbox).toBeEnabled();
        });
      });
      it('should have subtitle with text that reads Enabled', () => {
        render(<SocialShareWidget
          {...props}
          videoSharingEnabledForCourse
          allowVideoSharing={{
            level: 'block',
            value: false,
          }}
        />);
        expect(screen.getByText('Social Sharing')).toBeInTheDocument();
        fireEvent.click(screen.getByText('Social Sharing'));
        const subtitle = screen.getByText('Disabled');
        expect(subtitle).toBeInTheDocument();
      });
    });
  });
});
