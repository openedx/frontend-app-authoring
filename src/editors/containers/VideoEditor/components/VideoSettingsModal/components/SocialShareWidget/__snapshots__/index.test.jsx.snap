// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialShareWidget rendered with videoSharingEnabled false with default props should return null 1`] = `null`;

exports[`SocialShareWidget rendered with videoSharingEnabled false with videoSharingEnabledForAll false and isLibrary true should return null 1`] = `null`;

exports[`SocialShareWidget rendered with videoSharingEnabled false with videoSharingEnabledForCourse and isLibrary false and videoSharingEnabledForAll true should return null 1`] = `null`;

exports[`SocialShareWidget rendered with videoSharingEnabled true and allowVideoSharing value equals false should have subtitle with text that reads Enabled 1`] = `
<CollapsibleFormWidget
  fontSize="x-small"
  subtitle="Disabled"
  title="Social Sharing"
>
  <div>
    <FormattedMessage
      defaultMessage="Allow this video to be shareable to social media"
      description="Description for sociail sharing setting"
      id="authoring.videoeditor.socialShare.description"
    />
  </div>
  <Form.Checkbox
    checked={false}
    className="mt-3"
    disabled={false}
    onChange={[Function]}
  >
    <div
      className="small text-gray-700"
    >
      This video is shareable to social media
    </div>
  </Form.Checkbox>
  <div
    className="mt-3"
  >
    <Hyperlink
      className="text-primary-500"
      destination="sOMeURl.cOM"
      target="_blank"
    >
      Learn more about social sharing
    </Hyperlink>
  </div>
</CollapsibleFormWidget>
`;

exports[`SocialShareWidget rendered with videoSharingEnabled true and allowVideoSharing value equals true should have subtitle with text that reads Enabled 1`] = `
<CollapsibleFormWidget
  fontSize="x-small"
  subtitle="Enabled"
  title="Social Sharing"
>
  <div>
    <FormattedMessage
      defaultMessage="Allow this video to be shareable to social media"
      description="Description for sociail sharing setting"
      id="authoring.videoeditor.socialShare.description"
    />
  </div>
  <Form.Checkbox
    checked={true}
    className="mt-3"
    disabled={false}
    onChange={[Function]}
  >
    <div
      className="small text-gray-700"
    >
      This video is shareable to social media
    </div>
  </Form.Checkbox>
  <div
    className="mt-3"
  >
    <Hyperlink
      className="text-primary-500"
      destination="sOMeURl.cOM"
      target="_blank"
    >
      Learn more about social sharing
    </Hyperlink>
  </div>
</CollapsibleFormWidget>
`;
